### PR TITLE
Switch the order of env var for LWS_LEADER_ADDRESS

### PR DESCRIPTION
**Pull request by liurupeng**
#### What type of PR is this?
Quick change to make LWS_LEADER_ADDRESS to be the first container env var so we could reference that in the template from other env vars

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/utils/pod/pod_utils.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func addEnvVarIfNotExists(c *corev1.Container, e corev1.EnvVar) {
 			return
 		}
 	}
-	c.Env = append(c.Env, e)
+	c.Env = append([]corev1.EnvVar{e}, c.Env...)
```
Comment (kerthcet) [inline pkg/utils/pod/pod_utils.go:100]: Hostly this makes no sense to me as we can't guarantee the env orders, like once we add more envs, the `LWS_LEADER_ADDRESS` will not be the first one.

**Review by kerthcet — COMMENTED** [review]
What's the motivation here, can't get it. If the c.Env containers the `LWS_LEADER_ADDRESS`, I think we can get it from the slice although we have to iterate all its elements.

File: pkg/utils/pod/pod_utils.go (hunk 1 above)
Comment (liurupeng) [reply pkg/utils/pod/pod_utils.go:100]: yes, if there is another component modifying the env vars, it's very likely the first env var will not be LWS_LEADER_ADDRESS. The way that env var is resolved is sequential. So if I have an env var that want to use the value of leader address, then that env var need to be added after leader address env var

**Comment by liurupeng** [comment]
In general, I'd agree with a better approach which is to have an annotation to store the leader address

**Comment by kerthcet** [comment]
Talked offline, we want to put the internal env variables in prior to the use specified ones, then envs like `LWS_LEADER_ADDRESS` can be reused, then makes sense to me. Please add a comment to explain the reason. Thanks.

By the way, hope to use as less annotations as we can since we have a bunch of annotations already, my two cenets.

File: pkg/utils/pod/pod_utils.go (hunk 1 above)
Comment (ahg-g) [reply pkg/utils/pod/pod_utils.go:100]: I think it is fine, we are not promising anything.

**Pull request merged**